coap-gateway/service/session.go

```diff
@@ -303,21 +304,36 @@ func (c *session) onGetResourceContent(ctx context.Context, deviceID, href strin
 		return fmt.Errorf("cannot get resource /%v%v content: %w", deviceID, href, err)
 	}
 	notification.Hijack()
+	x := struct {
+		ctx                           context.Context
+		notification                  *pool.Message
+		deviceID                      string
+		href                          string
+		c                             *session
+		cannotGetResourceContentError func(deviceID, href string, err error) error
+	}{
+		ctx:                           ctx,
+		notification:                  notification,
+		deviceID:                      deviceID,
+		href:                          href,
+		c:                             c,
+		cannotGetResourceContentError: cannotGetResourceContentError,
+	}
 	err := c.server.taskQueue.Submit(func() {
-		defer c.server.messagePool.ReleaseMessage(notification)
-		if notification.Code() == codes.NotFound {
-			c.unpublishResourceLinks(c.getUserAuthorizedContext(ctx), []string{href}, nil)
+		defer x.c.server.messagePool.ReleaseMessage(x.notification)
+		if x.notification.Code() == codes.NotFound {
+			x.c.unpublishResourceLinks(x.c.getUserAuthorizedContext(x.ctx), []string{x.href}, nil)
 		}
-		err2 := c.notifyContentChanged(deviceID, href, false, notification)
+		err2 := x.c.notifyContentChanged(x.deviceID, x.href, false, x.notification)
 		if err2 != nil {
 			// hub is out of sync with the device, for recovery, the device is disconnected from the hub
-			c.Close()
-			c.Errorf("%w", cannotGetResourceContentError(deviceID, href, err2))
+			x.c.Close()
+			x.c.Errorf("%w", x.cannotGetResourceContentError(x.deviceID, x.href, err2))
 			return
 		}
-		obs, err := c.getDeviceObserver(c.Context())
+		obs, err := x.c.getDeviceObserver(x.c.Context())
```

Comment:
`err2` here so err doesn't get captured from the outside block?

Reply:
fixed